themes/default/content/blog/oidc-with-azure/index.md

```diff
@@ -0,0 +1,93 @@
+
+### Enabling OIDC
+
+There are two parts to enabling your Pulumi program to authenticate via OIDC. One is establishing the trust relationship between Azure and the other service that’s involved, like GitHub. This is a one-time operation. The second one is providing your program with the necessary configuration to perform the OIDC token exchange at runtime.
+
+Fortunately, Microsoft has the first step covered with their [establishing the trust relationship guide](https://learn.microsoft.com/en-us/azure/active-directory/workload-identities/workload-identity-federation-create-trust?pivots=identity-wif-apps-methods-azp). The idea is that an Active Directory app registration holds the necessary federated credentials. Your program will then use this app’s tenant id to request the token exchange. The setup depends on whether you run on GitHub Actions, Kubernetes, or other providers.
+
+On to the second step, configuring your Pulumi program. First, set the configuration `ARM_USE_OIDC` to true. Next, you’re in luck if your program runs on GitHub Actions: you're done. GitHub exports the necessary values in form of variables that Pulumi understands.
+
+{{% notes type="info" %}}
+For GitHub, pay close attention to the “entity type” of your Azure Active Directory credential as documented in the Azure guide. It can be a Git branch or tag, a pull request, or an environment as specified in the GitHub workflow. When your Action requests the OIDC token exchange, one of these needs to match exactly. This means that the only way to configure OIDC authentication for all branches or tags is to use environments in your workflows.
+{{% /notes %}}
+
+For other providers, you need to provide your Pulumi program with two more settings. `ARM_OIDC_REQUEST_TOKEN` is your provider’s token to exchange for an Azure token. `ARM_OIDC_REQUEST_URL` is the URL to contact to initiate the token exchange.
+
+### A complete example
+
+Let’s take a look at how the CI end-to-end tests of the Pulumi Azure Native provider are set up. First, we have an Active Directory App creatively named oidc-test. We simply copy its client and tenant ids and set them in the CI workflow, along with our subscription:
+
+```
+env:
+  ARM_CLIENT_ID: 30e520fa-[redacted]
+  ARM_SUBSCRIPTION_ID: 0282681f-[redacted]
+  ARM_TENANT_ID: 706143bc-[redacted]
+  ARM_USE_OIDC=true
+```
+
+Note how there is no secret or certificate present.
+
+We also give the required permissions to the workflow step where the authentication happens:
+
+```
+permissions:
+  id-token: write  # required for OIDC auth
+```
+
+In Azure, we added the required credentials to the Active Directory App. You'll find this under “Certificates & secrets” -> “Federated credentials” on the app’s page. Let’s look at two of the credentials. Both enable OIDC for all pull requests on the repository, one for each Azure provider.
+
+![OIDC credentials](oidc-credentials.png)
+
+That’s it! I hope this helps you simplify your Azure authentication.
```

Comment:
Is there a stronger call to action we can include here?

Reply:
Agreed. I reviewed the blog post and it looks great - but it ends a bit unexpectedly. Consider adding a brief section to the end with a wrap-up and a CTA.

Reply:
I tried to flesh out the final part a bit - let me know what you think.